cmd/create/cluster/cmd.go

```diff
@@ -1552,16 +1610,41 @@ func run(cmd *cobra.Command, _ []string) {
 	privateSubnetsCount := 0
 
 	var availabilityZones []string
+	var subnets []*ec2.Subnet
 	if useExistingVPC || subnetsProvided {
-		subnets, err := awsClient.GetSubnetIDs()
+		initialSubnets, err := awsClient.GetSubnetIDs()
 		if err != nil {
 			r.Reporter.Errorf("Failed to get the list of subnets: %s", err)
 			os.Exit(1)
 		}
 		if subnetsProvided {
 			useExistingVPC = true
 		}
+		_, machineNetwork, err := net.ParseCIDR(machineCIDR.String())
+		if err != nil {
+			r.Reporter.Errorf("Unable to parse machine CIDR")
+			os.Exit(1)
+		}
+		_, serviceNetwork, err := net.ParseCIDR(serviceCIDR.String())
+		if err != nil {
+			r.Reporter.Errorf("Unable to parse service CIDR")
+			os.Exit(1)
+		}
+		for _, subnet := range initialSubnets {
+			subnetIP, _, err := net.ParseCIDR(*subnet.CidrBlock)
+			if err != nil {
+				r.Reporter.Errorf("Unable to parse subnet CIDR")
+				os.Exit(1)
+			}
+
+			if machineNetwork.Contains(subnetIP) && !serviceNetwork.Contains(subnetIP) {
```

Comment:
For the service we check both sides

Intersect(n1, n2 *net.IPNet) bool {
	return n2.Contains(n1.IP) || n1.Contains(n2.IP)
}